### PR TITLE
fix: add cfg diagnostic for unresolved import error

### LINKED ISSUE #112674: cfg diagnostic not applied on imports
Import errors work differently than other path resolution errors and don't call `find_cfg_stripped` yet.

```rust
pub use a::x;

mod a {
    #[cfg(no)]
    pub fn x() {}
}

fn main() {}
```
this should note that `x` has been configured out.

### PR DESCRIPTION
**Pull request by bvanjoi**
Fixes #112674

An easy fix, r? @Nilstrieb

### CODE REVIEW

File: tests/ui/cfg/issue-112674.rs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+pub use a::x;
```
Comment (Nilstrieb) [inline tests/ui/cfg/issue-112674.rs:1]: can you rename the test to align it with the other tests for that diagnostic? either a new test with a similar name or just putting this into the existing tests

File: compiler/rustc_resolve/src/imports.rs [hunk 2]
```diff
@@ -667,6 +667,16 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                     _ => {}
                 }
             }
+
+            match &import.kind {
+                ImportKind::Single { source, .. } => {
+                    if let Some(ModuleOrUniformRoot::Module(module)) = import.imported_module.get()
+                    && let Some(module) = module.opt_def_id() {
```
Comment (Nilstrieb) [inline compiler/rustc_resolve/src/imports.rs:674]:
```suggestion
                    if let Some(ModuleOrUniformRoot::Module(module)) = import.imported_module.get()
                      && let Some(module) = module.opt_def_id()
                    {
```
rustfmt cant format let chains yet :(

File: tests/ui/cfg/diagnostics-reexport.stderr [hunk 3]
```diff
@@ -1,5 +1,17 @@
+error[E0432]: unresolved import `a::x`
+  --> $DIR/diagnostics-reexport.rs:12:9
+   |
+LL | pub use a::x;
+   |         ^^^^ no `x` in `a`
+   |
+note: found an item that was configured out
```
Comment (sladyn98) [inline tests/ui/cfg/diagnostics-reexport.stderr:7]: Not sure how difficult would it be to get it to say x has been configured out.

Comment (bvanjoi) [reply tests/ui/cfg/diagnostics-reexport.stderr:7]: It had listed below

**Review by Noratrieb — COMMENTED** [review]
could you add one more test for
```rust
pub use b::{x,y};
//~^ ERROR unresolved imports `b::x`, `b::y`
//~| NOTE no `x` in `b`
//~| NOTE no `y` in `b`

mod b {
    #[cfg(no)]
    pub fn x() {}
    //~^ NOTE found an item that was configured out
    #[cfg(no)]
    pub fn y() {}
    //~^ NOTE found an item that was configured out
}
```
r=me aftwerwards

**Pull request merged**